packages/paypal-commerce-integration/src/paypal-commerce-credit-card/paypal-commerce-credit-cards-payment-strategy.ts

```diff
@@ -345,56 +372,30 @@ export default class PayPalCommerceCreditCardsPaymentStrategy implements Payment
         };
     }
 
-    private getCardTypeByEvent({
-        cards,
-    }: PayPalCommerceHostedFieldsState): HostedFieldCardTypeChangeEventData {
-        return {
-            cardType: cards?.[0]?.type,
-        };
-    }
-
     /**
      *
      * Hosted form submit method
      *
      * */
-    private async submitHostedForm(methodId: string): Promise<string> {
-        const hostedFields = this.getHostedFieldsOrThrow();
-
-        const state = this.paymentIntegrationService.getState();
-        const paymentMethod = state.getPaymentMethodOrThrow(methodId);
-        const { is3dsEnabled } = paymentMethod.config;
-
-        const options: PayPalCommerceHostedFieldsSubmitOptions = {
-            ...(this.cardNameField?.value && {
-                cardholderName: this.cardNameField.value,
-            }),
-            ...(is3dsEnabled && {
-                contingencies: ['3D_SECURE'],
-            }),
-        };
-
-        const { liabilityShift, orderId } = await hostedFields.submit(options);
+    private async submitHostedForm() {
+        const cardFields = this.getCardFieldsOrThrow();
 
-        if (is3dsEnabled && (liabilityShift === 'NO' || liabilityShift === 'UNKNOWN')) {
-            // FIXME: we should throw another error to have an ability to translate it
+        await cardFields.submit().catch(() => {
```

Comment:
I tried to use `try ... catch ...` here, but for some reason it doesn't work because of specific `submit` PayPal method. Perhaps they've created some custom `Promise` which works only this way.